### PR TITLE
Fix nodes-readiness to take into account the default number of edge replicas when the zones array is filled

### PR DESCRIPTION
**Pull request by aleskandro**

### CODE REVIEW

**Comment by r4f4** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-edge-zones

File: ci-operator/step-registry/nodes/readiness/nodes-readiness-commands.sh [hunk 1]
```diff
@@ -54,9 +54,10 @@ function wait_for_nodes_readiness()
   done
 }
 
+EDGE_ZONES_COUNT=$(yq -r '.compute[] | select(.name == "edge") | .platform.aws.zones | length // 0' "${SHARED_DIR}/install-config.yaml")
```
Comment (r4f4) [inline ci-operator/step-registry/nodes/readiness/nodes-readiness-commands.sh:57]:
```suggestion
EDGE_ZONES_COUNT=$(yq-v4 -r '.compute[] | select(.name == "edge") | .platform.aws.zones | length // 0' "${SHARED_DIR}/install-config.yaml")
```

**Comment by aleskandro** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-edge-zones

File: ci-operator/step-registry/nodes/readiness/nodes-readiness-commands.sh [hunk 2]
```diff
@@ -54,9 +54,10 @@ function wait_for_nodes_readiness()
   done
 }
 
+EDGE_ZONES_COUNT=$(yq-v4 -r '.compute[] | select(.name == "edge") | .platform.aws.zones | length // 0' "${SHARED_DIR}/install-config.yaml")
 expected_nodes=$(( $(yq-v4 -r '.controlPlane.replicas // '"${CONTROL_PLANE_REPLICAS:-3}" "${SHARED_DIR}/install-config.yaml") +
   $(yq-v4 -r '.compute[] | select(.name == "worker") | .replicas // '"${COMPUTE_NODE_REPLICAS:-3}" "${SHARED_DIR}/install-config.yaml") +
-  $(yq-v4 -r '.compute[] | select(.name == "edge") | .replicas // '"${EDGE_WORKER_NUMBER:-0}" "${SHARED_DIR}/install-config.yaml")
+  $(yq-v4 -r '.compute[] | select(.name == "edge") | .replicas // '"${EDGE_WORKER_NUMBER:-${EDGE_ZONES_COUNT})}" "${SHARED_DIR}/install-config.yaml")
```
Comment (r4f4) [inline ci-operator/step-registry/nodes/readiness/nodes-readiness-commands.sh:60]:
```suggestion
  $(yq-v4 -r '.compute[] | select(.name == "edge") | .replicas // '"${EDGE_WORKER_NUMBER:-${EDGE_ZONES_COUNT}}" "${SHARED_DIR}/install-config.yaml")
```

**Comment by aleskandro** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-edge-zones

**Comment by aleskandro** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-edge-zones

**Comment by r4f4** [comment]
It's fixed now!

/lgtm

**Comment by aleskandro** [comment]
/pj-rehearse ack

```
INFO[2024-08-08T14:32:41Z] Running step e2e-aws-ovn-edge-zones-nodes-readiness. 
INFO[2024-08-08T14:32:52Z] Step e2e-aws-ovn-edge-zones-nodes-readiness succeeded after 11s. 
```

**Pull request merged**